app/src/androidTest/java/in/testpress/testpress/fragment/DeleteDownloadedVideosTest.kt

```diff
@@ -0,0 +1,59 @@
+package `in`.testpress.testpress.fragment
+
+import `in`.testpress.testpress.authenticator.LoginActivity
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+import androidx.test.core.app.ApplicationProvider.getApplicationContext
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.ActivityTestRule
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class DeleteDownloadedVideosTest {
+
+    private val testActivityRule = ActivityTestRule(LoginActivity::class.java, true, true)
+
+    @Rule
+    fun rule() = testActivityRule
+
+    private lateinit var context: Context
+
+    private var usernamePref: String? = null
+
+    @Before
+    fun setup() {
+        context = getApplicationContext()
+        val sharedPreferences = context.getSharedPreferences(
+                "UserPreference",
+                MODE_PRIVATE
+        )
+        usernamePref = sharedPreferences.getString("username", "")
+    }
+
+    @Test
+    fun whenUsernameIsDifferentVideosShouldDelete() {
+        val result = testActivityRule.activity.deleteOfflineVideosForDifferentAccount("test")
```

Comment:
Here you are directly checking for username "test", where is logging in step as previous user